docs/proposals/20210203-unmanaged-infrastructure.md

```diff
@@ -0,0 +1,235 @@
+
+This predicate filters out any resource that has been marked as "externally managed" and prevents the controller from reconciling the resource.
+
+### Security Model
+
+When externally managed, the required cloud provider privileges required by CAPI might be significantly reduced when compared with a traditionally managed cluster.
+The only privileges required by CAPI are those that are required to manage machines.
+
+For example, when an AWS cluster is managed by CAPI, persmissions are required to be able to create VPCs and other networking components that are managed by the AWSCluster controller. When externally managed, these permissions are not required as the external entity is reponsible for creating such components.
+
+### Risks and Mitigations
+
+#### What happens when a user converts an externally managed InfraCluster to a managed InfraCluster?
+
+There currently is no immutability support for CRD annotations within the Kubernetes API.
+
+This means that, once a user has created their externally managed InfraCluster, they could at some point, update the annotation to make the InfraCluster appear to be managed.
+
+There is no way to predict what would happen in this scenario.
+The InfraCluster controller would start attempting to reconcile infrastructure that it did not create, and therefore, there may be assumptions it makes that mean it cannot manage this infrastructure.
+
+To prevent this, we will have to implement (in the InfraCluster webhook) a means to prevent users converting externally managed InfraClusters into managed InfraClusters.
+
+Note however, converting from managed to externally managed should cause no issues and should be allowed.
+It will be documented as part of the externally managed contract that this is a one way operation.
+
+### Future Work
+
+#### Marking InfraCluster ready manually
+
+The content of this proposal assumes that the management of the external infrastructure is done by some controller which has the ability to set the spec and status of the InfraCluster resource.
+
+In reality, this may not be the case. For example, if the infrastructure was created by an admin using Terraform.
+
+When using a system such as this, a user can copy the details from the infrastructure into an InfraCluster resource and create this manually.
+However, they will not be able to set the InfraCluster to ready as this requires updating the resource status which is difficult when not using a controller.
+
+To allow users to adopt this external management pattern without the need for writing their own controllers or tooling, we will provide a longer term solution that allows a user to indicate that the infrastructure is ready and have the status set appropriately.
+
+The exact mechanism for how this will work is undecided, though the following ideas have been suggested:
+
+- Write a kubectl plugin that allows a user to mark their InfraCluster resources as ready
```

Comment:
Given @nikhita 's POC and KEP on adding support for modifying status and scale subresources, I think we can update this and cross-link?

Reply:
sg but I can't find it. Would you mind sharing it?

Reply:
https://github.com/kubernetes/kubernetes/pull/99556 was the POC. Nikhita's taking user stories in the comments there to then convert to a KEP.

Reply:
thanks! updated.